pokemongo_bot/walkers/step_walker.py

```diff
@@ -8,147 +8,108 @@
-                'last_position': (self.initLat, self.initLng, self.initAlt),
-                'distance': '',
-                'distance_unit': ''
-            }
-        )
         self.bot.heartbeat()
 
-        sleep(1)  # sleep one second plus a random delta
-        # self._work_at_position(
-        #     self.initLat, self.initLng,
-        #     alt, False)
+        return self.is_arrived()
+
+    def is_arrived(self):
+        return distance(self.bot.position[0], self.bot.position[1], self.dest_lat, self.dest_lng) <= self.precision
 
     def _calc_bearing(self, start_lat, start_lng, dest_lat, dest_lng):
         """
         Calculates the bearing between two points.
-    
+
         The formulae used is the following:
             θ = atan2(sin(Δlong).cos(lat2),
                       cos(lat1).sin(lat2) − sin(lat1).cos(lat2).cos(Δlong))
-    
+
         :Parameters:
           - `start_lat in decimal degrees
           - `start_lng in decimal degrees
           - `dest_lat in decimal degrees
           - `dest_lng in decimal degrees
-    
+
         :Returns:
           The bearing in degrees
-    
+
         :Returns Type:
           float
         """
-    
+
         lat1 = math.radians(start_lat)
         lat2 = math.radians(dest_lat)
-    
+
         diffLong = math.radians(dest_lng - start_lng)
-    
+
         x = math.sin(diffLong) * math.cos(lat2)
-        y = math.cos(lat1) * math.sin(lat2) - (math.sin(lat1)
-                * math.cos(lat2) * math.cos(diffLong))
-    
+        y = math.cos(lat1) * math.sin(lat2) - (math.sin(lat1) * math.cos(lat2) * math.cos(diffLong))
+
         initial_bearing = math.atan2(x, y)
-    
+
         # Now we have the initial bearing but math.atan2 return values
         # from -180° to + 180° which is not what we want for a compass bearing
         # The solution is to normalize the initial bearing as shown below
         initial_bearing = math.degrees(initial_bearing)
         compass_bearing = (initial_bearing + 360) % 360
-    
+
         return compass_bearing
-        
-    def _get_next_pos(self, lat, lon, bearing, speed, precision):
-        origin = Point(lat, lon)
-        if speed == 0.0:
-            return lat, lon
+
+    def get_next_position(self, origin_lat, origin_lng, origin_alt, dest_lat, dest_lng, dest_alt, speed):
+        origin = Point(origin_lat, origin_lng, origin_alt)
+
+        total_distance = distance(origin_lat, origin_lng, dest_lat, dest_lng)
+
+        if total_distance == 0:
+            total_distance = self.precision
+
         if speed == float("inf"):
-            return self.destLat, self.destLng
+            dist = total_distance
         else:
-            offset_angle = (1/self.speed)*(precision/1.74)
-            lat, lon, _ = VincentyDistance(kilometers=speed*1e-3).destination(origin, bearing + uniform(-offset_angle, offset_angle))
-            return lat, lon
+            dist = max(min(total_distance, speed), self.precision)
```

Comment:
offset_angle when total_distance = 0 is  off

In [17]: (100/0.5)/(0.5/1.74) % 360
Out[17]: 336.0

that is about 0.5m away from origin and will start spiralling with a 336 bearing 

```
In [32]: haversine.haversine((0,0), tuple(VincentyDistance(kilometers=0.5 * 1e-3).destination(Point(0,0), 0 + (100/0.5)/(0.5/1.74) % 360)[0:2]))*1000
Out[32]: 0.5022511946715152

In [33]: 
```